test/framework/src/main/java/org/elasticsearch/ingest/TestIngestDocument.java

```diff
@@ -8,30 +8,32 @@
 
 package org.elasticsearch.ingest;
 
+import org.elasticsearch.core.Tuple;
+import org.elasticsearch.index.VersionType;
+import org.elasticsearch.test.ESTestCase;
+
 import java.util.HashMap;
 import java.util.Map;
+import java.util.function.BiConsumer;
 
 /**
  * Construct ingest documents for testing purposes
  */
 public class TestIngestDocument {
 
     /**
-     * These two test static factory methods are needed for testing and allow to the creation of a new {@link IngestDocument} given the
-     * provided elasticsearch metadata, source and ingest metadata.
-     *
-     * This is needed because the ingest metadata will be initialized with the current timestamp at init time, which makes equality
-     * comparisons impossible in tests.
+     * Create an IngestDocument for testing that pass an empty mutable map for ingestMetaata
      */
-    public static IngestDocument ofSourceAndIngest(Map<String, Object> sourceAndMetadata, Map<String, Object> ingestMetadata) {
-        return new IngestDocument(sourceAndMetadata, ingestMetadata);
+    public static IngestDocument ofSourceAndMetadata(Map<String, Object> sourceAndMetadata) {
+        return new IngestDocument(sourceAndMetadata, new HashMap<>());
```

Comment:
Doesn't this need to be the metadata from the sourceAndMetadata?

Reply:
No, there's a separate IngestMetadata for internal processor use.